### PR TITLE
vscode_extension_installer: use UID for installs

### PR DESCRIPTION
**Pull request by Bo98**
This should make sense in the general case. VSCode should always install based on the calling user rather than the middle man.

### CODE REVIEW

File: lib/bundle/vscode_extension_installer.rb [hunk 1]
```diff
@@ -24,13 +24,36 @@ def preinstall(name, no_upgrade: false, verbose: false)
...
+
     def install(name, preinstall: true, no_upgrade: false, verbose: false, force: false)
       return true unless preinstall
       return true if extension_installed?(name)
 
       puts "Installing #{name} VSCode extension. It is not currently installed." if verbose
 
-      return false unless Bundle.system "code", "--install-extension", name, verbose: verbose
+      return false unless exchange_uid do
+        Bundle.system("code", "--install-extension", name, verbose:)
```
Comment (jacobbednarz) [inline lib/bundle/vscode_extension_installer.rb:55]: missing the `verbose` value here

**Review by jacobbednarz — CHANGES_REQUESTED** [review]

File: lib/bundle/vscode_extension_installer.rb (hunk 1 above)
Comment (MikeMcQuaid) [reply lib/bundle/vscode_extension_installer.rb:55]: @jacobbednarz Sorry, you're mistaken: this is newer, valid Ruby syntax.

Comment (jacobbednarz) [reply lib/bundle/vscode_extension_installer.rb:55]: TIL! explains why my old Ruby marked this as a syntax error. thanks!

**Review by MikeMcQuaid — APPROVED** [review]

**Comment by MikeMcQuaid** [comment]
Thanks again @Bo98!

**Pull request merged**